go/libraries/doltcore/sqlserver/dolt_server.go

```diff
@@ -15,31 +15,56 @@
 package sqlserver
 
 import (
+	"fmt"
 	"sync"
 
+	"github.com/dolthub/dolt/go/libraries/doltcore/env"
+
 	"github.com/dolthub/go-mysql-server/server"
 )
 
-var mySQLServer *server.Server
-var mySQLServerMutex sync.Mutex
+var lockedDetails *serverAndLockfile
+var mutex sync.Mutex
+
+// Struct for holding a *server.Server and a *env.DBLock
```

Comment:
(minor) godocs should start with the identifier (`serverAndLockfile` here) 

```suggestion
// serverAndLockfile holds a *server.Server and a *env.DBLock for a running server 
```